pkg/ccl/importccl/exportparquet.go

```diff
@@ -181,57 +210,356 @@ func newParquetColumn(typ *types.T, name string, nullable bool) (parquetColumn,
+		schemaEl.LogicalType.TIME.IsAdjustedToUTC = true // per crdb docs
 
-	case types.FloatFamily:
-		if typ.Oid() == oid.T_float8 {
-			col.definition.SchemaElement.Type = parquet.TypePtr(parquet.Type_DOUBLE)
-			col.encodeFn = func(d tree.Datum) (interface{}, error) {
-				return float64(*d.(*tree.DFloat)), nil
+		col.encodeFn = func(d tree.Datum) (interface{}, error) {
+			// Time of day is stored in microseconds since midnight,
+			// which is also how parquet stores time
+			time := d.(*tree.DTime)
+			m := int64(*time)
+			return m, nil
+		}
+		col.DecodeFn = func(x interface{}) (tree.Datum, error) {
+			return tree.MakeDTime(timeofday.TimeOfDay(x.(int64))), nil
+		}
+	case types.TimeTZFamily:
+		// The parquet vendor does not support an efficient encoding of TimeTZ
+		// (i.e. a datetime field and a timezone field), so we must fall back to
+		// encoding the whole TimeTZ as a string.
+		populateLogicalStringCol(schemaEl)
+		col.encodeFn = func(d tree.Datum) (interface{}, error) {
+			return []byte(d.(*tree.DTimeTZ).TimeTZ.String()), nil
+		}
+		col.DecodeFn = func(x interface{}) (tree.Datum, error) {
+			d, dependsOnCtx, err := tree.ParseDTimeTZ(nil, string(x.([]byte)), time.Microsecond)
+			if dependsOnCtx {
+				return nil, fmt.Errorf("parsed ime depends on context")
+			}
+			return d, err
+		}
+	case types.IntervalFamily:
+		// The parquet vendor actually supports intervals as a parquet converted type,
+		// but converted types have been deprecated in the Apache Parquet format.
+		// https://github.com/fraugster/parquet-go#supported-converted-types
+		populateLogicalStringCol(schemaEl)
+		col.encodeFn = func(d tree.Datum) (interface{}, error) {
+			return []byte(d.(*tree.DInterval).ValueAsISO8601String()), nil
+		}
+		col.DecodeFn = func(x interface{}) (tree.Datum, error) {
+			return tree.ParseDInterval(duration.IntervalStyle_ISO_8601, string(x.([]byte)))
+		}
+	case types.TimestampFamily:
+		// Didn't encode this as Microseconds ( original attempt commented out)
+		// since the unix epoch because of threat of overflow. See comment
+		// associated with time.Time.UnixMicro().
+		populateLogicalStringCol(schemaEl)
+
+		/*schemaEl.Type = parquet.TypePtr(parquet.Type_INT64)
+		  schemaEl.LogicalType = parquet.NewLogicalType()
+		  schemaEl.LogicalType.TIMESTAMP = parquet.NewTimestampType()
+		  t := parquet.NewTimeUnit()
+		  t.MICROS = parquet.NewMicroSeconds()
+		  schemaEl.LogicalType.TIMESTAMP.Unit = t
+		  schemaEl.LogicalType.TIMESTAMP.IsAdjustedToUTC = true*/
+
+		col.encodeFn = func(d tree.Datum) (interface{}, error) {
+			ts := roundtripStringer(d.(*tree.DTimestamp))
+			return []byte(ts), nil
+		}
+		col.DecodeFn = func(x interface{}) (tree.Datum, error) {
+			//return tree.MakeDTimestamp(time.UnixMicro(x.(int64)).UTC(), time.Microsecond)
+			dtStr := string(x.([]byte))
+			d, dependsOnCtx, err := tree.ParseDTimestamp(nil, dtStr, time.Microsecond)
+			if dependsOnCtx {
+				return nil, fmt.Errorf("TimestampTZ depends on context")
```

Comment:
Are we sure we want these to be errors? I only ask because they aren't in some of the other decoders.

Reply:
are you looking at any other decoders aside from avro.go in changeefeeds? the avro writer/reader hooked into avro.go avoids this because it can be fed time.time types while the parquet vendor cannot.

Reply:
I was looking at the csv reader which I believe calls https://github.com/cockroachdb/cockroach/blob/f1fcb8303cac913074f7feecf461ea513b3e8e9f/pkg/sql/rowenc/roundtrip_format.go#L21 which appears to ignore the dependsOnCtx return, but I'll admit I only took a cursory glance.  Since this is currently only used in tests, this isn't a major issue either way.